cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/azure/azure-dev/cli/azd/pkg/tools/docker"
 )
 
-type dockerCredentials struct {
+type DockerCredentials struct {
```

Comment:
Would we want to provide some explanation of the newly exported type here to explain how this is both a "traditional username password" but could also be an "AAD token credential"?